spaceinvaders.py

```diff
@@ -132,9 +137,12 @@ def update(self, current_time):
                     self.rightMoves = 30 + self.leftAddMove
                 self.direction *= -1
                 self.moveNumber = 0
+                self.bottom = 0
                 for enemy in self:
-                    enemy.rect.y += 35
+                    enemy.rect.y += ENEMY_MOVE_DOWN
                     enemy.toggle_image()
+                    if self.bottom < enemy.rect.y + 35:
+                        self.bottom = enemy.rect.y + 35
```

Comment:
Shouldn't the game be over if the enemies reach the bottom? Is that what this is doing?

Reply:
No, this strings just calculate the bottom line of enemies. Game over happens in `check_collisions`:
```
if not self.player.alive() or self.enemies.bottom >= 600:
    self.gameOver = True
```